app/Services/StandService.php

```diff
@@ -542,46 +544,110 @@ private function standOccupied(NetworkAircraft $aircraft, Stand $stand): bool
+        )->unique('callsign');
+    }
+
+    /**
+     * Find aircraft that do not have an arrival stand assigned and are eligible.
+     * Additional criteria are:
+     *
+     * 1. Must not already have an assigned stand.
+     */
+    private function getAircraftWithoutAStandAssignmentThatRequireOneForArrival(): Collection
+    {
+        return $this->getBaseArrivalAllocationQuery()
+            ->leftJoin(
+                'stand_assignments',
+                'network_aircraft.callsign',
+                '=',
+                'stand_assignments.callsign'
+            )
+            ->whereNull('stand_assignments.callsign')
+            ->get();
     }
 
-    public function getAircraftEligibleForArrivalStandAllocation(): Collection
+    /**
+     * Find aircraft that already have an arrival stand, but we need to renew it.
+     * Additional criteria are:
+     *
+     * 1. The stand assigned is neither at their destination or departure airfield, which indicates
+     * a diversion scenario.
+     */
+    private function getAircraftThatHaveArrivalStandAllocationsButRequireRenewing(): Collection
     {
-        return NetworkAircraft::whereIn(
-            'planned_destairport',
-            Airfield::all()->pluck('code')->toArray()
+        return $this->getBaseArrivalAllocationQuery()->join(
+            'stand_assignments',
+            'network_aircraft.callsign',
+            '=',
+            'stand_assignments.callsign'
         )
-            ->whereDoesntHave('assignedStand')
+            ->join('stands', 'stand_assignments.stand_id', '=', 'stands.id')
+            ->join('airfield', 'stands.airfield_id', '=', 'airfield.id')
+            ->whereRaw('airfield.code <> network_aircraft.planned_destairport')
+            ->whereRaw('airfield.code <> network_aircraft.planned_depairport')
             ->get();
     }
 
+    /**
+     * Get the base query for determining if an aircraft is eligible for arrival stand allocation.
+     * The criteria are:
+     *
+     * 1. Must be a known aircraft type
+     * 2. Must be a type of aircraft for which stands are assigned
+     * 3. Must be arriving into an airport that we care about
+     * 4. Must not be arriving into the same airport as departure airport (because circuits)
+     * 5. Must be moving, we can't calculate time until arrival if they're not moving.
+     */
+    private function getBaseArrivalAllocationQuery(): Builder
+    {
+        return NetworkAircraft::join('aircraft', 'aircraft.code', '=', 'network_aircraft.planned_aircraft')
+            ->where('aircraft.allocate_stands', '=', 1)
+            ->join('airfield AS arrival_airfield', 'network_aircraft.planned_destairport', '=', 'arrival_airfield.code')
+            ->whereRaw('network_aircraft.planned_depairport <> network_aircraft.planned_destairport')
+            ->where('network_aircraft.groundspeed', '>', 0)
+            ->select('network_aircraft.*');
```

Comment:
Do we need to select all columns here?

Reply:
Various columns get used (and in future may get used!) in this bit - so having a fully populated model is helpful